### ISSUE TITLE
Villager trades won't progress and professions are off from trades given and other bugs.

### ISSUE THREAD

**Issue by Catscrath25** [WildBamaBoy/minecraft-comes-alive#938]
I'm running MC 1.12.2 and it's just as it says in the title. I was looking for a cartographer and noticed that there were only farmers and fisherman in the village despite there being a library. I did some more searching for the day and found a weapon smith with farmer trades and the guards had off trades as well like fetchers and fishermen. The sun was setting and I didn't yet have a bed because I spent the day looking for the map maker so I decided to hold up in the library. Once night came and all the villagers teleported to their homes I was greeted by a fisherman who lived there. I tried his trade only to found out that HE was the map maker. So I gave him several stacks of paper to unlock further trade and got three emeralds. I exited his trade and waited for the particle effects... they never happened to I entered his trade again. To my surprise he still wasn't accepting any more paper but didn't have any new trades. I'm pretty sure this is not how you wanted this to work so I'm bringing it to you XD. Hopefully you can fix this soon and I can go on my treasure hunts and temple raids soon. Thanks ahead of time! Great mod all the way!

**Comment by Catscrath25** [comment]
I have also found that MCA villagers will become infected to be vanilla villagers. Curing them turns them back to an MCA villager but they are not the same as when they were infected (Different name, gender, profession and trades.) as well as being invisible.

**Comment by Galindorf** [comment]
Ok so, the mismatched trades thing is straight up just a thing in this version. But this might help you: If you told him to "stay here" so he will be safe, it sort of "freezes" them and they won't progress trades. Tell him to "move freely" and THEN trade with him.

Also, don't try to edit their profession to match their trade type (using the villager editor)... it changes their trades too. I now have a librarian who has librarian trades for her 1st 2 trades (that I unlocked), I used the villager editor to change her profession to Librarian, and her next tier trades are now Butcher trades. No magic books for me! :<
